backend/layers/thirdparty/s3_provider.py

```diff
@@ -80,6 +81,15 @@ def delete_files(self, bucket_name: str, object_keys: List[str]) -> None:
                 logger.info(f"Errors: {errors}")
                 raise S3DeleteException(errors)
 
+    def delete_recursive(self, bucket_name: str, prefix: str) -> None:
```

Comment:
You don't need a new function. Use`list_directory` and `delete_files` to accomplish this, or use those two function in `delete_recursive`

Reply:
The prefix delete is going to be way faster; if you have to iterate over the dozens of individual s3 keys associated with a single cxg file, it'll increase network time costs a lot.